### DISCUSSION TITLE
Output from SD card to A2DP and I2S

### DISCUSSION THREAD

**Discussion by gary7530** [pschatzmann/arduino-audio-tools#598] [Q&A] [answered]
I am now using the examples "player-sdfat-i2s" and "player-sdfat-a2dp", I want to merge the two examples together

                                         
SD card -> MultiOutput -> a2dp and i2s

**Comment by pschatzmann** [comment]
I would expect that you just need to define the MultiOutput as output for the Player.

**Comment by gary7530** [comment]
I want to read the MP3 from the SD card in ESP1 and send it to ESP2 with A2DP to play the music, and ESP1 will play the music from the SD card synchronously.

ESP1 and ESP2 will play the same music at the same time

**Comment by pschatzmann** [comment]
This will be pretty tricky: The A2DP adds a big delay to the output, so though you might be playing at the same time you will not hear them at the same time!

**Reply by gary7530** [reply]
I want to make Multiroom audio, do you have any ideas or suggestions?

**Reply by pschatzmann** [reply]
- You can still try your initial idea but you need to have an ESP32 with SPI-RAM so that you can handle the time difference with a simple Ringbuffer
- Use 1 ESP32 as Player that distributes mp3 data via ESP-Now and receive the data by different ESP32 in each room which decode and play the received data 

I never tried something like this so I don't know how well the played audio will be in synch...

**Reply by gary7530** [reply]
Do you have a sample program for ESP-Now Multiroom audio?

**Comment by pschatzmann** [comment, marked as answer]
No, but an example that shows 
- how to use ESPNow: https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-communication/esp-now
- how to use the player to output mp3 to a destination stream: https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-vs1053/player-sd-vs1053/player-sd-vs1053.ino
